backend/src/core/server.ts

```diff
@@ -50,26 +40,21 @@ export default class Server {
 
   private setupSockets(): void {
     const io = socketIo(this.httpServer);
-    io.on(Event.CONNECT, socket => {
+    io.on(Event.CONNECT, (socket: ExtendedSocket) => {
+      socket.username = "";
```

Comment:
Why is this necessary? If it is to satisfy the Typescript type in some way maybe we could change it to be optional (i.e. `username?: string` in the `ExtendedSocket` interface

Reply:
You're right I made the username optional and it satisfied the Typescript typing thank you